components/DropdownMenu.vue

```diff
@@ -56,14 +56,14 @@ function menuItemClick(event: Event, item: DropdownMenuItem) {
         </p>
         <p
           v-else
-          @click="(event: MouseEvent) => event.stopPropagation()"
           class="flex justify-start items-center gap-1 py-2 px-3"
+          @click="(event: MouseEvent) => event.stopPropagation()"
         >
           <IconComponent v-if="item.icon" :name="item.icon" />
           <span>{{ item.text }}</span>
         </p>
       </li>
     </ul>
-    <div @click="close" class="fixed inset-0 z-10"></div>
+    <div class="fixed inset-0 z-10" @click="close"></div>
```

Comment:
Can we use a button here instead of a div? Since we have a click handler here it makes sense to use the semantic HTML element